azurerm/internal/services/apimanagement/api_management_certificate_resource.go

```diff
@@ -44,15 +46,30 @@ func resourceApiManagementCertificate() *schema.Resource {
 
 			"data": {
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
 				Sensitive:    true,
 				ValidateFunc: validation.StringIsBase64,
 			},
```

Comment:
could we add the RequiredWith & conflicts with property here too? it needs to be on both properties it applies to

Reply:
Added conflicts with on data. Both key_vault_secret_id and data can be set by themselves so shouldn't need a RequiredWith on them.